catalog/app/containers/Admin/Buckets.tsx

```diff
@@ -290,125 +445,120 @@ function BucketFields({ add = false, reindex }) {
-    file_extensions_to_index: get(
-      'fileExtensionsToIndex',
-      R.pipe(
-        R.split(','),
-        R.map(R.trim),
-        R.reject((t) => !t),
-        R.uniq,
-      ),
-    ),
-    scanner_parallel_shards_depth: get('scannerParallelShardsDepth', Number),
-    sns_notification_arn: get('snsNotificationArn', (v) => {
-      if (v === DO_NOT_SUBSCRIBE_SYM) return DO_NOT_SUBSCRIBE_STR
-      if (typeof v === 'string') return v.trim()
-      return undefined
-    }),
-    skip_meta_data_indexing: get('skipMetaDataIndexing'),
-    delay_scan: get('delayScan'),
+const ADD_MUTATION = `
+  mutation Admin_Buckets_Add($input: BucketAddInput!) {
+    bucketAdd(input: $input) {
+      ... on BucketAddSuccess {
+        bucketConfig {
+          name
+          title
+          iconUrl
+          description
+          relevanceScore
+          overviewUrl
+          tags
+          linkedData
+          fileExtensionsToIndex
+          scannerParallelShardsDepth
+          snsNotificationArn
+          skipMetaDataIndexing
+          lastIndexed
+        }
+      }
+    }
   }
-  return R.reject(isMissing, json)
+`
+
+interface AddData {
+  bucketAdd: Model.BucketAddResult
 }
 
-const toBucketConfig = (b) => ({
-  name: b.name,
-  title: b.title,
-  iconUrl: b.iconUrl || DEFAULT_ICON,
-  description: b.description,
-  overviewUrl: b.overviewUrl,
-  linkedData: b.linkedData,
-  tags: b.tags,
-  relevance: b.relevanceScore,
-})
-
-function useAuthSession() {
-  const cfg = Config.use()
-  const sessionId = redux.useSelector(AuthSelectors.sessionId)
-  return cfg.alwaysRequiresAuth && sessionId
+interface AddVariables {
+  input: Model.BucketAddInput
 }
 
-function Add({ close }) {
-  const req = APIConnector.use()
-  const cache = Cache.use()
-  const session = useAuthSession()
+interface AddProps {
+  close: (reason?: string) => void
+}
+
+function Add({ close }: AddProps) {
   const { push } = Notifications.use()
   const t = useTracker()
+  const [, add] = urql.useMutation<AddData, AddVariables>(ADD_MUTATION)
   const onSubmit = React.useCallback(
     async (values) => {
       try {
-        const res = await req({
-          endpoint: '/admin/buckets',
-          method: 'POST',
-          body: JSON.stringify(formToJSON(values)),
-        })
-        const b = data.bucketFromJSON(res)
-        cache.patchOk(data.BucketsResource, null, R.append(b))
-        cache.patchOk(
-          BucketConfig.BucketsResource,
-          { empty: false, session },
-          R.append(toBucketConfig(b)),
-          true,
-        )
-        push(`Bucket "${b.name}" added`)
-        t.track('WEB', { type: 'admin', action: 'bucket add', bucket: b.name })
-        close()
-      } catch (e) {
-        if (APIConnector.HTTPError.is(e, 409, /Bucket already added/)) {
-          throw new RF.SubmissionError({ name: 'conflict' })
+        const input = R.applySpec(addFormSpec)(values)
+        const res = await add({ input })
```

Comment:
Maybe use `Admin/requests.ts` and place this request there?

Reply:
that seems like a step backwards, bc
1. graphql provides a clean way to define requests without much boilerplate (i.e. converting data types back and forth and issuing a fetch call or smth), so everything you need to do to define a request is writing a graphql mutation (and a type annotation for it bc currently we dont have automatic type generation or anything like that);
2. i believe that code should be grouped by feature, not by "type" (so, for example, request definitions should be close to the code making these requests, not in a single place shared by the whole app), meaning we shouldnt put all admin requests in one place, bc they dont have anything in common (all the "common" / standardized parts are already in graphql schema and `model` module);